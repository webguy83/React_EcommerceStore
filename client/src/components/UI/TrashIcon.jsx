import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';

import styled from 'styled-components/macro';

const TrashIconContainer = styled.span`
    cursor: pointer;

    &:hover, &:active {
        color: var(--sec);
        transition: var(--btnTrans);
    }
`

const TrashIcon = ({ removeItem }) => {
    return (
        <TrashIconContainer onClick={removeItem}>
            <FontAwesomeIcon icon={faTrashAlt} />
        </TrashIconContainer>
    );
};

export default TrashIcon;