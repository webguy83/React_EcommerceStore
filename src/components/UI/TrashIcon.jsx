import React from 'react';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';

import styled from 'styled-components/macro';

const TrashIconContainer = styled.span`
    &:hover {
        cursor: pointer;
        color: #F4B400;
        transition: all .3s ease-in-out;
    }
`

const TrashIcon = ({ removeItem }) => {
    return (
        <TrashIconContainer onClick={removeItem}>
            <FontAwesomeIcon icon={faTrashAlt} />
        </TrashIconContainer>
    );
};

export default TrashIcon;