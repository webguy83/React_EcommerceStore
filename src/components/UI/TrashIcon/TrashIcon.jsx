import React from 'react';

import { TrashIconContainer } from './TrashIcon.module.scss';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';

const TrashIcon = ({ removeItem }) => {
    return (
        <span className={TrashIconContainer} onClick={removeItem}>
            <FontAwesomeIcon icon={faTrashAlt} />
        </span>
    );
};

export default TrashIcon;