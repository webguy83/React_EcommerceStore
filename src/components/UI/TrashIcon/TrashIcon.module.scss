@import '../../../mixins/UI.scss';

.TrashIconContainer {
    &:hover {
        @include buttonHover;
    }
}