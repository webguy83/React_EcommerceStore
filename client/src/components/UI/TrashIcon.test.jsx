import React from "react";
import TrashIcon from "./TrashIcon";
import "jest-styled-components";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";

describe("default styles for TrashIcon", () => {
  it("it should render with no errors", () => {
    const { asFragment } = render(<TrashIcon removeItem={() => {}} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
