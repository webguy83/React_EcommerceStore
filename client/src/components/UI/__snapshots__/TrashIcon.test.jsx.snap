// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default styles for TrashIcon it should render with no errors 1`] = `
<DocumentFragment>
  .c0 {
  cursor: pointer;
  font-size: 2.9rem;
}

.c0:hover,
.c0:active {
  color: var(--sec);
  -webkit-transition: var(--btnTrans);
  transition: var(--btnTrans);
}

<span
    class="c0"
  >
    ðŸ—‘
  </span>
</DocumentFragment>
`;
